src/themes/uclouvain/app/entity-groups/publication-entity/item-pages/publication-page-citations/publication-page-citations.component.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { ItemCitationsService } from '../../citations/item-citations.service';
+import { isNotEmpty } from 'src/app/shared/empty.util';
+import { Item } from 'src/app/core/shared/item.model';
+import { BehaviorSubject, distinctUntilChanged, filter, finalize, map, Observable, of, Subscription, switchMap, take, tap } from 'rxjs';
+import { ItemCitation } from 'src/app/core/shared/item-citations.model';
+import { NotificationsService } from 'src/app/shared/notifications/notifications.service';
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
+import { ItemExportFormatMolteplicity, ItemExportFormatService } from 'src/app/core/itemexportformat/item-export-format.service';
+import { ItemExportFormat, ItemExportFormatMap } from 'src/app/core/itemexportformat/model/item-export-format.model';
+import { AsyncPipe, NgClass, NgFor, NgIf } from '@angular/common';
+
+/**
+ * Component to display all the available citations for a given publication item.
+ * You can select a format to see the related citation and copy it to the clipboard.
+ * 
+ * @author Michaël Pourbaix (michael.pourbaix@uclouvain.be)
+ */
+@Component({
+  selector: 'ds-publication-citations',
+  templateUrl: './publication-page-citations.component.html',
+  styleUrls: ['./publication-page-citations.component.scss'],
+  standalone: true,
+  imports: [
+    NgIf,
+    AsyncPipe,
+    TranslateModule,
+    NgFor,
+    NgClass,
+  ]
+})
+export class PublicationPageCitationsComponent implements OnInit, OnDestroy {
+  @Input() dso: Item;
+  @Input() selectFirst: boolean = true;
+
+  protected readonly isNotEmpty = isNotEmpty;
+
+  protected loadingCitation: BehaviorSubject<boolean> = new BehaviorSubject(false);
+  protected loadingFormats: BehaviorSubject<boolean> = new BehaviorSubject(false);
+
+  protected citationFormats$: Observable<ItemExportFormat[]> = of(null)
+
+  protected citationContent$: Observable<ItemCitation> = of(null);
+  protected selectedCitationFormat$: BehaviorSubject<ItemExportFormat> = new BehaviorSubject(null);
+
+  protected subs: Subscription[] = [];
+
+  constructor(
+    protected itemCitationsService: ItemCitationsService,
+    protected notificationService: NotificationsService,
+    protected translateService: TranslateService,
+    protected itemExportFormatService: ItemExportFormatService,
+  ) { }
+
+  ngOnInit(): void {
+    if (isNotEmpty(this.dso) && this.dso.id) {
```

Comment:
strange to the if `@Input dso` is not empty here... This component shoudn't be called if `dso` is null